framework/tlvf/src/include/tlvf/CmduMessage.h

```diff
@@ -52,10 +53,21 @@ class CmduMessage {
     void swap() { msg.swap(); };
     bool is_finalized() const { return msg.is_finalized(); };
     bool is_swapped() const { return msg.is_swapped(); };
+    void set_max_tlv_length(size_t length) { m_max_tlv_length = length; };
+    size_t max_tlv_length() { return m_max_tlv_length; };
 
     static const size_t kMaxCmduLength      = 1500;
     static const uint16_t kCmduHeaderLength = 8;
     static const uint16_t kTlvHeaderLength  = 3;
+
+private:
+    // By default, we assume that the CMDU will be sent over the ieee1905_1 transport.
+    // Since the ieee1905_1 transport only supports CMDU fragmantation on TLVs boundaries,
+    // a TLV max size can't be larger than the MTU - CMDU header length - EOM TLV length.
+    // However, when the CMDU is sent internally via UDS, this restriction does not apply.
+    // So we allo changing it via set_max_tlv_length() API.
+    size_t m_max_tlv_length =
+        kMaxCmduLength - kCmduHeaderLength - ieee1905_1::tlvEndOfMessage::get_initial_size();
```

Comment:
I'm sorry but this whole commit is featuring a bad solution IMO. Limiting CMDU size to MTU and creating special cases for "jumbo" tlvs is not an elegant solution.
There should not be any size limit expect the defined by the given buffer size, which is:
`buff_len - cmdu_header_len - tlv_end_of_message_len'

Limiting the TLV size should not be handled in the tlvf, it should be handled in the transport layer.
If we have exceeded MTU size, we should get that failure on the BTL or the transport when trying to send the CMDU.

Reply:
The ieee1905 defines fragmantation to be based on TLV boundary, so each TLV must never exceed the MTU - CMDU header length.
Adding the check here is also necessary since we don't know if this CMDU will be transmitted over the transport.

> Limiting the TLV size should not be handled in the tlvf, it should be handled in the transport layer.

Correct - and it is handled there, which is how this was reproduced in the first place - there is an error in the transport when trying to send oversized TLVs - see `ieee1905_transport_packet_processing.cpp::fragment_and_send_packet_to_network_interface()` function.

We can't know in advance if the TLV should be sent over to the network, in which case it will fail if it has an oversized TLV. The best solution is never to send a TLV which is larger then `MTU - CMDU header length - TLV end of message length` - but this is problematic since `network_map` minimum node size is already bigger than 2KB.

The solution of using the jumbo TLVs is the most elegant I could think of, I'm sorry you don't like it, but I think that for now we should take it as is since this PR is already huge and took too long to be merged, we can refactor that in the future if we decide to do so.

Reply:
> We can't know in advance if the TLV should be sent over to the network, in which case it will fail if it has an oversized TLV. The best solution is never to send a TLV which is larger then MTU - CMDU header length - TLV end of message length - but this is problematic since network_map minimum node size is already bigger than 2KB.

You don't need to know it in advance. You should get an error when trying to send such TLV to the bus. So if such an error exists on the transport, then the only change you can do in this PR is to set the tlv limitation to `MTU - CMDU header length - TLV end of message length` and remove the mumbo jumbo commits.

Reply:
Moran - Well done, I just misunderstood you but now with the resize solution in place there is no longer any need to limit the TLV size, so I removed the relevant commits. Thanks for finding that!